language/move-lang/src/typing/expand.rs

```diff
@@ -180,6 +202,50 @@ fn exp(context: &mut Context, e: &mut T::Exp) {
                 Kind_::Unknown | Kind_::Affine | Kind_::Resource => E::Move { from_user, var },
             }
         }
+        E::InferredNum(v) => {
+            use BaseType_ as B;
+            use BuiltinTypeName_ as BT;
+            use SingleType_ as S;
+            use TypeName_ as TN;
+            use Type_ as T;
+            macro_rules! builtin {
+                ($b:pat) => {
+                    T::Single(sp!(
+                        _,
+                        S::Base(sp!(_, B::Apply(_, sp!(_, TN::Builtin(sp!(_, $b))), _)))
+                    ))
+                };
+            }
+            let bt = match &e.ty.value {
+                builtin!(bt) if bt.is_numeric() => bt,
+                _ => panic!("ICE inferred num failed {:?}", &e.ty.value),
+            };
+            let v = *v;
+            let max = match bt {
+                BT::U8 => std::u8::MAX as u128,
+                BT::U64 => std::u64::MAX as u128,
+                BT::U128 => std::u128::MAX,
+                _ => unreachable!(),
+            };
+            if v > max {
+                let msg = format!(
+                    "Expected a literal of type '{}', but the value is too large",
+                    bt
+                );
+                context.error(vec![
+                    (e.exp.loc, "Invalid numerical literal".into()),
+                    (e.ty.loc, msg),
+                ]);
+                return;
+            }
```

Comment:
What do we do in the case where the number `v` is expressed as a value greater than `u128::MAX` in the source? Is this checked in parsing?

Reply:
It would error in parsing